### ISSUE TITLE
Failure to build OMR error: cannot find symbol generated.OmrBuildFlags

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#9126]
Failure is intermittent.

https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_OMR/536/
- this one passed on restart https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_OMR/537/

https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_linux_cm_Nightly/335
```
22:50:50  Compiling 2 files for BUILD_DDR_STUBS
22:50:50  Compiling 5029 files for BUILD_J9DDR_MAIN_CLASSES
22:51:24  /home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_Nightly/openj9/debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/ObjectMonitor_V1.java:41: error: cannot find symbol
22:51:24  import com.ibm.j9ddr.vm29.pointer.generated.OmrBuildFlags;
22:51:24                                             ^
22:51:24    symbol:   class OmrBuildFlags
22:51:24    location: package com.ibm.j9ddr.vm29.pointer.generated
22:51:24  /home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_Nightly/openj9/debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/ObjectMonitor_V2.java:41: error: cannot find symbol
22:51:24  import com.ibm.j9ddr.vm29.pointer.generated.OmrBuildFlags;
22:51:24                                             ^
22:51:24    symbol:   class OmrBuildFlags
22:51:24    location: package com.ibm.j9ddr.vm29.pointer.generated
etc.
```

**Comment by pshipton** [comment]
@keithc-ca @dnakamura

**Comment by keithc-ca** [comment]
I suspect an important clue is that in the failing build there are numerous warnings like the sample below, but none in the successful one.
```
16:45:01  Dependee "/home/jenkins/workspace/Build_JDK8_x86-64_linux_cm_OMR/build/linux-x86_64-normal-server-release/vm/sourcetools/j9nls/CMakeFiles/j9nls.dir/DependInfo.cmake" is newer than depender "/home/jenkins/workspace/Build_JDK8_x86-64_linux_cm_OMR/build/linux-x86_64-normal-server-release/vm/sourcetools/j9nls/CMakeFiles/j9nls.dir/depend.internal".
```
I'd guess that this is because parallelism is not being managed properly with respect to cmake build generation; that is generation is occurring more often than expected.

**Comment by dnakamura** [comment]
I've seen a lot of "depender newer than dependee" warnings /w jdk8. IIRC, they are harmless warnings that are only emitted, because jdk8 makefiles specify some kind of verbose flag (issue does not show up with jdk9+)

**Comment by keithc-ca** [comment]
Just because those warnings are suppressed in jdk11+, doesn't mean we shouldn't pay attention to them.

Has anyone seen this locally? (the diagnostics archives attached to those builds are empty)

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_OMR/541

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_linux_cm_OMR/549/

**Comment by pshipton** [comment]
This still occurs https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_Nightly/362

@DanHeidinga fyi. I expect this intermittent problem needs to be resolved before cmake is enabled for any platform.

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_Nightly/363

**Comment by DanHeidinga** [comment]
@dnakamura I've listed this in #9195 as a must fix for the cmake rollout.  Can you comment here and I'll assign it to you?

**Comment by dnakamura** [comment]
Looking into the issue, I have some theories.

**Comment by dnakamura** [comment]
This one is proving difficult to track down. Even remoting into a ci machine which has failures, I can't get the issue to reproduce, even though it seems to happen fairly reliably in ci builds. My command line is copy pasted straight out of the jenkins log, and I'm not seeing any obvious differnces in the shell environment

**Comment by dnakamura** [comment]
I'm thinking this may be some machine weirdness. Does not seem to occur on internal builds. Can't reproduce even on same machine with identical SHAs. The most obvious answer would be not tracking a dependency properly in make, causing a race condition. However in this case the logs show that ddrgen processes the debuginfo/macros long before we generate the java code. Current plan of attack is to set up jenkins to capture some more useful diagnostic info if this should show up again

**Comment by pshipton** [comment]
It was still occurring, although I haven't seen one since last week.
All the build machines should be the same, the last few were on different machines.

https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_OMR/575 - cent6-x64-5
https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_OMR/574 - cent6-x64-3
https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_OpenJDK8/40 - cent6-x64-7

**Comment by pshipton** [comment]
Just a reminder this problem still occurs, although I don't report all the failures.
https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_OMR/600

**Comment by dnakamura** [comment]
@pshipton  have you noticed any obvious patterns? Based on a quick look, it appears to be more common on the jdk 8 builds.

**Comment by pshipton** [comment]
> it appears to be more common on the jdk 8 builds

This seems true, but since I haven't collected all the failures it's hard to say for sure.

**Comment by keithc-ca** [comment]
I don't see how #9670 can 'fix' this: reopening.

**Comment by dnakamura** [comment]
My bad. github saw the keyword "resolve" next to an issue number, so merging the pr causes it to mark the issue as resolved

**Comment by pshipton** [comment]
Another occurrence https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_Nightly/399
which should have the extra diagnostics @dnakamura enabled. @keithc-ca fyi

**Comment by dnakamura** [comment]
It looks like a combination of hitting a timeout in a command and not properly detecting the timeout as an error condition. Given the fact that this seems isolated to jdk8 (or at least much more frequent there), I'm suspecting its just some combination of our messing with the jobserver mechanics and ddrgen being called in a different location in the makefiles.

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_Nightly/408

**Comment by keithc-ca** [comment]
> https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_cm_Nightly/408

In that build, `build/linux-x86_64-normal-server-release/vm/runtime/omr/omrddr/omrddr/__/build/linux-x86_64-normal-server-release/vm/runtime/omr/omrcfg.annt.h` is empty. It should contain:
```
@DDRFILE_BEGIN .../vm/runtime/omr/omrcfg.h
@TYPE_OmrBuildFlags
@MACRO_OMR_GC 1
@MACRO_OMR_GC_DEFINED 1
@MACRO_OMR_COMPILER 1
@MACRO_OMR_COMPILER_DEFINED 1
@MACRO_OMR_JITBUILDER 0
@MACRO_OMR_JITBUILDER_DEFINED 0
@MACRO_OMR_TEST_COMPILER 0
@MACRO_OMR_TEST_COMPILER_DEFINED 0
@MACRO_OMR_PORT 1
@MACRO_OMR_PORT_DEFINED 1
...
```

**Comment by dnakamura** [comment]
@pshipton  has this popped up again? I believe it should be fixed with eclipse/omr#5266

**Comment by pshipton** [comment]
I haven't seen it.
